diff --git a/node_modules/@evershop/s3_file_storage/dist/services/awsFileBrowser.js b/node_modules/@evershop/s3_file_storage/dist/services/awsFileBrowser.js
index 09edceb..bec8a45 100644
--- a/node_modules/@evershop/s3_file_storage/dist/services/awsFileBrowser.js
+++ b/node_modules/@evershop/s3_file_storage/dist/services/awsFileBrowser.js
@@ -1,6 +1,11 @@
 import { S3Client, ListObjectsV2Command } from "@aws-sdk/client-s3";
 import { getEnv } from "@evershop/evershop/lib/util/getEnv";
-const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+// Patched for Yandex S3 *: endpoint + forcePathStyle
+const s3Client = new S3Client({
+    region: getEnv("AWS_REGION"),
+    endpoint: getEnv("AWS_ENDPOINT"),
+    forcePathStyle: true
+  });
 const bucketName = getEnv("AWS_BUCKET_NAME");
 export const awsFileBrowser = {
     list: async (path) => {
@@ -23,7 +28,8 @@ export const awsFileBrowser = {
             ? data.Contents.filter((item) => item.Size !== 0).map((object) => {
                 var _a;
                 const fileName = (_a = object.Key) === null || _a === void 0 ? void 0 : _a.split("/").pop();
-                const fileURL = `https://${bucketName}.s3.amazonaws.com/${object.Key}`;
+               // Patched for Yandex S3 *
+                const fileURL = `${getEnv("AWS_ENDPOINT")}/${bucketName}/${object.Key}`;
                 return {
                     name: fileName,
                     url: fileURL,
diff --git a/node_modules/@evershop/s3_file_storage/dist/services/awsFileDeleter.js b/node_modules/@evershop/s3_file_storage/dist/services/awsFileDeleter.js
index 30a7d64..1f8b906 100644
--- a/node_modules/@evershop/s3_file_storage/dist/services/awsFileDeleter.js
+++ b/node_modules/@evershop/s3_file_storage/dist/services/awsFileDeleter.js
@@ -1,6 +1,11 @@
 import { S3Client, DeleteObjectCommand, HeadObjectCommand, } from "@aws-sdk/client-s3";
 import { getEnv } from "@evershop/evershop/lib/util/getEnv";
-const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+// Patched for Yandex S3 *: endpoint + forcePathStyle
+const s3Client = new S3Client({
+    region: getEnv("AWS_REGION"),
+    endpoint: getEnv("AWS_ENDPOINT"),
+    forcePathStyle: true
+});
 const bucketName = getEnv("AWS_BUCKET_NAME");
 export const awsFileDeleter = {
     delete: async (path) => {
diff --git a/node_modules/@evershop/s3_file_storage/dist/services/awsFileUploader.js b/node_modules/@evershop/s3_file_storage/dist/services/awsFileUploader.js
index d5f3923..24f18bf 100644
--- a/node_modules/@evershop/s3_file_storage/dist/services/awsFileUploader.js
+++ b/node_modules/@evershop/s3_file_storage/dist/services/awsFileUploader.js
@@ -1,7 +1,14 @@
 import path from "path";
 import { S3Client, PutObjectCommand, } from "@aws-sdk/client-s3";
 import { getEnv } from "@evershop/evershop/lib/util/getEnv";
-const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+// Patched for Yandex S3 *: endpoint + forcePathStyle
+const s3clientparams = { 
+    region: getEnv("AWS_REGION"),
+    endpoint: getEnv("AWS_ENDPOINT"),
+    forcePathStyle: true
+}
+const s3Client = new S3Client(s3clientparams);
+
 const bucketName = getEnv("AWS_BUCKET_NAME");
 export const awsFileUploader = {
     upload: async (files, requestedPath) => {
@@ -27,7 +34,8 @@ export const awsFileUploader = {
                 name: files[index].filename,
                 mimetype: files[index].mimetype,
                 size: files[index].size,
-                url: `https://${bucketName}.s3.amazonaws.com/${path.join(requestedPath, files[index].filename)}`,
+                // Patched for Yandex S3 *
+                url: `${getEnv("AWS_ENDPOINT")}/${bucketName}/${path.posix.join(requestedPath ?? "", files[index].filename)}`,
             });
         });
         return uploadedFiles;
diff --git a/node_modules/@evershop/s3_file_storage/dist/services/awsFolderCreator.js b/node_modules/@evershop/s3_file_storage/dist/services/awsFolderCreator.js
index 642c6ad..54f0157 100644
--- a/node_modules/@evershop/s3_file_storage/dist/services/awsFolderCreator.js
+++ b/node_modules/@evershop/s3_file_storage/dist/services/awsFolderCreator.js
@@ -1,6 +1,11 @@
 import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
 import { getEnv } from "@evershop/evershop/lib/util/getEnv";
-const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+// Patched for Yandex S3 *: endpoint + forcePathStyle
+const s3Client = new S3Client({
+    region: getEnv("AWS_REGION"),
+    endpoint: getEnv("AWS_ENDPOINT"),
+    forcePathStyle: true
+});
 const bucketName = getEnv("AWS_BUCKET_NAME");
 export const awsFolderCreator = {
     create: async (path) => {
diff --git a/node_modules/@evershop/s3_file_storage/dist/subscribers/product_image_added/awsGenerateProductImageVariant.js b/node_modules/@evershop/s3_file_storage/dist/subscribers/product_image_added/awsGenerateProductImageVariant.js
index 977ba54..e8c9919 100644
--- a/node_modules/@evershop/s3_file_storage/dist/subscribers/product_image_added/awsGenerateProductImageVariant.js
+++ b/node_modules/@evershop/s3_file_storage/dist/subscribers/product_image_added/awsGenerateProductImageVariant.js
@@ -9,14 +9,22 @@ import { error } from "@evershop/evershop/lib/log";
 async function downloadObjectToBuffer(objectUrl) {
     var _a;
     const parsedUrl = new URL(objectUrl);
-    const bucketName = parsedUrl.host.split(".")[0]; // Extract the bucket name
-    const objectKey = parsedUrl.pathname.substr(1); // Extract the object key (remove leading '/')
+    const bucketName = getEnv("AWS_BUCKET_NAME"); // Extract the bucket name for Yandex S3 *
+    // Extract the object key (remove leading '/' and bucket segment) for Yandex S3 *
+    const pathParts = parsedUrl.pathname.replace(/^\/+/, "").split("/");
+    pathParts.shift();
+    const objectKey = pathParts.join("/");
     const params = {
         Bucket: bucketName,
         Key: objectKey,
     };
     const getObjectCommand = new GetObjectCommand(params);
-    const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+    const s3clientparams = { 
+      region: getEnv("AWS_REGION"),
+      endpoint: getEnv("AWS_ENDPOINT"),
+      forcePathStyle: true
+  } 
+  const s3Client = new S3Client(s3clientparams);
     const data = await s3Client.send(getObjectCommand);
     // Get content as a buffer from the data.Body object
     const buffer = await ((_a = data.Body) === null || _a === void 0 ? void 0 : _a.transformToByteArray());
@@ -31,7 +39,10 @@ async function resizeAndUploadImage(s3Client, originalObjectUrl, resizedObjectUr
         .toBuffer();
     // Upload the resized image
     const parsedUrl = new URL(resizedObjectUrl);
-    const objectKey = parsedUrl.pathname.substr(1); // Extract the object key (remove leading '/')
+    // Extract the object key (remove leading '/' and bucket segment) for Yandex S3 *
+    const uploadPathParts = parsedUrl.pathname.replace(/^\/+/, "").split("/");
+    uploadPathParts.shift();
+    const objectKey = uploadPathParts.join("/");
     const uploadParams = {
         Bucket: bucketName,
         Key: objectKey,
@@ -43,12 +54,17 @@ async function resizeAndUploadImage(s3Client, originalObjectUrl, resizedObjectUr
 }
 export default async function awsGenerateProductImageVariant(data) {
     if (getConfig("system.file_storage") === "s3") {
-        try {
-            const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+    try {
+                const s3clientparams = { 
+                  region: getEnv("AWS_REGION"),
+                  endpoint: getEnv("AWS_ENDPOINT"),
+                  forcePathStyle: true
+              }
+              const s3Client = new S3Client(s3clientparams);
             const originalObjectUrl = data.origin_image;
             // The data.image is the full url of the Object, we need to get the Object path
             // by removing the container url
-            const ext = path.extname(originalObjectUrl);
+            const ext = path.extname(originalObjectUrl)
             // Target path for single variant by adding a '-single' just before the extension
             const singleObjectUrl = originalObjectUrl.replace(ext, `-single${ext}`);
             // Target path for listing variant by adding a '-listing' just before the extension
diff --git a/node_modules/@evershop/s3_file_storage/src/services/awsFileBrowser.ts b/node_modules/@evershop/s3_file_storage/src/services/awsFileBrowser.ts
index b3f38d7..8d7bb42 100644
--- a/node_modules/@evershop/s3_file_storage/src/services/awsFileBrowser.ts
+++ b/node_modules/@evershop/s3_file_storage/src/services/awsFileBrowser.ts
@@ -27,7 +27,7 @@ export const awsFileBrowser = {
     const files = data.Contents
       ? data.Contents.filter((item) => item.Size !== 0).map((object) => {
           const fileName = object.Key?.split("/").pop();
-          const fileURL = `https://${bucketName}.s3.amazonaws.com/${object.Key}`;
+                const fileURL = `${getEnv("AWS_ENDPOINT")}/${bucketName}/${object.Key}`;
 
           return {
             name: fileName,
diff --git a/node_modules/@evershop/s3_file_storage/src/services/awsFileDeleter.ts b/node_modules/@evershop/s3_file_storage/src/services/awsFileDeleter.ts
index 8563cef..ccb2852 100644
--- a/node_modules/@evershop/s3_file_storage/src/services/awsFileDeleter.ts
+++ b/node_modules/@evershop/s3_file_storage/src/services/awsFileDeleter.ts
@@ -5,7 +5,10 @@ import {
 } from "@aws-sdk/client-s3";
 import { getEnv } from "@evershop/evershop/lib/util/getEnv";
 
-const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+const s3Client = new S3Client({ region: getEnv("AWS_REGION"),
+  endpoint: getEnv("AWS_ENDPOINT"),
+  forcePathStyle: true
+});
 const bucketName = getEnv("AWS_BUCKET_NAME");
 
 export const awsFileDeleter = {
diff --git a/node_modules/@evershop/s3_file_storage/src/subscribers/product_image_added/awsGenerateProductImageVariant.ts b/node_modules/@evershop/s3_file_storage/src/subscribers/product_image_added/awsGenerateProductImageVariant.ts
index 80fc2e2..ef28b80 100644
--- a/node_modules/@evershop/s3_file_storage/src/subscribers/product_image_added/awsGenerateProductImageVariant.ts
+++ b/node_modules/@evershop/s3_file_storage/src/subscribers/product_image_added/awsGenerateProductImageVariant.ts
@@ -22,7 +22,11 @@ async function downloadObjectToBuffer(objectUrl: string) {
   };
 
   const getObjectCommand = new GetObjectCommand(params);
-  const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+  const s3Client = new S3Client({ 
+    region: getEnv("AWS_REGION"),
+    endpoint: getEnv("AWS_ENDPOINT"),
+    forcePathStyle: true
+  });
   const data = await s3Client.send(getObjectCommand);
   // Get content as a buffer from the data.Body object
   const buffer = await data.Body?.transformToByteArray();
@@ -59,13 +63,23 @@ async function resizeAndUploadImage(
 }
 
 export default async function awsGenerateProductImageVariant(data) {
+  console.log('data (FROM SRC)', data);
   if (getConfig("system.file_storage") === "s3") {
+    console.log('getConfig("system.file_storage") === "s3"**', true);
     try {
-      const s3Client = new S3Client({ region: getEnv("AWS_REGION") });
+      const s3clientparams = { 
+        region: getEnv("AWS_REGION"),
+        endpoint: getEnv("AWS_ENDPOINT"),
+        forcePathStyle: true
+    }
+    console.log('s3clientparams', s3clientparams);
+    const s3Client = new S3Client(s3clientparams);
+      console.log('s3Client', s3Client);
       const originalObjectUrl = data.origin_image;
       // The data.image is the full url of the Object, we need to get the Object path
       // by removing the container url
       const ext = path.extname(originalObjectUrl);
+      console.log('ext', ext);
       // Target path for single variant by adding a '-single' just before the extension
       const singleObjectUrl = originalObjectUrl.replace(ext, `-single${ext}`);
       // Target path for listing variant by adding a '-listing' just before the extension
@@ -103,6 +117,10 @@ export default async function awsGenerateProductImageVariant(data) {
         getConfig("catalog.product.image.thumbnail.height", 100)
       );
 
+      console.log('singleUrl', singleUrl);
+      console.log('listingUrl', listingUrl);
+      console.log('thumnailUrl', thumnailUrl);
+
       // Update the record in the database with the new URLs in the variant columns
       await update("product_image")
         .given({
